modules/github_integration/lib/api/v3/github_integration/github_pull_request_representer.rb

```diff
@@ -0,0 +1,166 @@
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) 2012-2021 the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See docs/COPYRIGHT.rdoc for more details.
+#++
+
+require 'roar/decorator'
+require 'roar/json/hal'
+
+module API
+  module V3
+    module GithubIntegration
+      class GithubPullRequestRepresenter < ::API::Decorators::Single
+        include API::Caching::CachedRepresenter
+        include API::Decorators::DateProperty
+        include API::Decorators::LinkedResource
+
+        cached_representer key_parts: %i[github_user merged_by latest_check_runs],
+                           disabled: false
+
+        link :staticPath do
+          next
+        end
```

Comment:
??? -> this is a Noop as far as I can tell.

Reply:
it leaves me puzzled too, without it I get the following error, but only if the cached_representer is set to `disabled: false` (not when `disabled: true`).

<img width="1027" alt="Screenshot 2021-04-12 at 22 57 01" src="https://user-images.githubusercontent.com/206108/114461742-67b8a380-9be2-11eb-8e39-28a696108796.png">

Reply:
it was hard to specify an actual static path since we live in the scope of a work package (so that we know the project to check permissions against).

This is because the `:show_github_content` permission is set on a project-level, but a github-pr may link to many work packages across many projects. This makes it hard (impossible?) for me to decide on a global scope whether it's OK to show the PR or not

Reply:
I think we haven't had the case yet, where a representer without a link is to be cached. This should work of course but it also points out how this representer deviates from how the v3 is working (with a lot of links). 

To solve this problem independent from the permission discussion, please add a `self` link as it is needed anyway. It might be as simple as adding a `self_link title_getter: ->(*) { title }`

Reply:
<img width="1005" alt="Screenshot 2021-04-13 at 08 52 20" src="https://user-images.githubusercontent.com/206108/114509313-a6c41480-9c35-11eb-8483-9859789f04e1.png">

`

Reply:
a fabolous pairing session with @ulferts resolved this